docs/source/spec/http-protocol-compliance-tests.rst

```diff
@@ -0,0 +1,534 @@
+
+* Timestamp values must be converted to a Unix timestamp represented
+  as an integer.
+* Client implementations that automatically provide values for members marked
+  with the :ref:`idempotencyToken-trait` MUST use a constant value of
+  ``00000000-0000-4000-8000-000000000000``.
+
+
+.. _httpRequestTests-trait:
+
+----------------
+httpRequestTests
+----------------
+
+Summary
+    The ``httpRequestTests`` trait is used to define how an HTTP request is
+    serialized given a specific protocol, authentication scheme, and set of
+    input parameters.
+Trait selector
+    .. code-block:: css
+
+        operation
+Value type
+    ``object``
+
+The ``httpRequestTests`` trait is an object that supports the following
+properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 25 65
+
+    * - Property
+      - Type
+      - Description
+    * - id
+      - string
+      - **Required**. The identifier of the test case. This identifier can
+        be used by protocol test implementations to filter out unsupported
+        test cases by ID, to generate test case names, etc. The provided
+        ``id`` MUST match Smithy's :token:`identifier` ABNF. No two
+        ``httpRequestTests`` test cases can share the same ID.
+    * - protocol
+      - string
+      - **Required**. The name of the :ref:`protocol <protocols-trait>` to
+        test. Because Smithy services can support multiple protocols, each
+        test MUST specify which protocol is under test.
+    * - method
+      - string
+      - **Required**. The expected serialized HTTP request method.
+    * - uri
+      - string
+      - **Required**. The request-target of the HTTP request, not including
+        the query string (for example, "/foo/bar").
+    * - authScheme
+      - string
+      - The optional :ref:`authentication scheme <auth-trait>` to assume.
+        It's possible that specific authentication schemes might influence
+        the serialization logic of an HTTP request.
+    * - queryParams
+      - Map<String, String>
+      - A map of expected query string parameters.
+
+        A serialized HTTP request is not in compliance with the protocol
+        if any query string parameter defined in ``queryParams`` is not
+        defined in the request or if the value of a query string parameter
+        in the request differs from the expected value.
+
+        Each key represents the query string parameter name, and each
+        value represents the query string parameter value. Both keys and
+        values MUST appear in the format in which it is expected to be
+        sent over the wire; if a key or value needs to be percent-encoded,
+        then it MUST appear percent-encoded in this map.
+
+        ``queryParams`` applies no constraints on additional request headers.
+    * - forbidQueryParams
+      - [string]
+      - A list of query string parameter names that must not appear in the
+        serialized HTTP request.
```

Comment:
Missing this from the implementation docs:

```
A serialized HTTP request is not in compliance with the protocol
if any query parameters listed in `forbidQueryParams` are present
in the serialized request.
```

We usually allow the implementation documentation to be less verbose than the docs, but not vice-versa.

Reply:
Removed from .smithy model since it's overly verbose and restating the previous sentence.